src/readKongApi.js

```diff
@@ -106,10 +106,15 @@ export function parseRoute({ id, created_at, updated_at, service, ...rest }, ser
         const route = (config.services.find((s) => s.name === serviceName).routes || [])
               .find((r) => r.id === id);
         if (route) {
-            name = route.name;
+            // Prior to Kong 1.0, routes didn't have names, but the kongfig config format allowed a route
+            // to specify a name as a top level property (as a sibling to attributes). Kong 1.0 added names
+            // as an actual property of a route, so it should be specified as attributes.name. For
+            // backwards compatibility, kongfig still uses the top level name property, but will prefer
+            // attributes.name.
+            name = route.attributes.name || route.name;
         }
     }
-    return { name: (name || id), id, attributes: {...rest}, _info: { id, updated_at, created_at } };
+    return { name, id, attributes: {...rest}, _info: { id, updated_at, created_at } };
```

Comment:
From the comment above:

> Kong 1.0 added names as an actual property of a route, so it should be specified as attributes.name ...

So should name be next to `...rest`, inside `attributes`?

Reply:
Turns out this was not a wise choice 😅. Adding name to attributes here prevented kongfig from detecting when the name in kong didn't match the name in the config.